zircon.core/src/commonMain/kotlin/org/hexworks/zircon/api/builder/application/AppConfigBuilder.kt

```diff
@@ -191,6 +191,25 @@ data class AppConfigBuilder(
         )
     }
 
+    /**
+     * Adds a custom property into the AppConfig object. This can later be retrieved using [AppConfig.getProperty].
+     *
+     * ### End Developers
+     *
+     * You probably don't need to call this API.
+     *
+     * ### Plugin Developers
+     *
+     * Write extension methods off of [AppConfigBuilder] that call this API in order to enable end developers
```

Comment:
This is an awesome idea. Would you add a small example for this here?

Reply:
> We might decide at some point that this is ugly and Java users should learn Kotlin instead. :)

I for one would welcome not using Maybes :smile: since Kotlin annotates nullable things as nullable anyway, Java developers should still see warnings crop up in their IDE, at least.

> small example

I'll see if I can get a `@sample` annotation working. Documentation-oriented test case probably.